jerry-core/lit/lit-char-helpers.c

```diff
@@ -458,6 +462,207 @@ lit_char_is_word_char (ecma_char_t c) /**< code unit */
+      {
+        output_buffer_p[0] = is_lowercase ? character : (ecma_char_t) (current_sp + char_dist - 1);
+      }
+
+      return 1;
+    }
+
+    if (character > current_sp)
+    {
+      bottom = middle + 1;
+    }
+    else
+    {
+      top = middle - 1;
+    }
+  }
+
+  return 0;
+} /* search_in_character_pair_ranges */
+
+/**
+ * Check if the specified character is part of the character_pairs table.
+ *
+ * @return the mapped character sequence of an ecma character, if it's in the table.
+ *         0 - otherwise.
+ */
+static ecma_length_t
+search_in_character_pairs (ecma_char_t character,        /**< code unit */
+                           ecma_char_t *output_buffer_p, /**< [out] buffer for the result characters */
+                           bool is_lowercase)            /**< is lowercase conversion */
+{
+  int num_of_character_pairs = NUM_OF_ELEMENTS (jerry_character_pairs);
+
+  for (int i = 0; i < num_of_character_pairs; i++)
+  {
+    if (character != jerry_character_pairs[i])
+    {
+      continue;
+    }
+
+    if (i % 2 == 0)
+    {
+      output_buffer_p[0] = is_lowercase ? jerry_character_pairs[i + 1] : character;
+    }
+    else
+    {
+      output_buffer_p[0] = is_lowercase ? character : jerry_character_pairs[i - 1];
+    }
+
+    return 1;
+  }
+
+  return 0;
+} /* search_in_character_pairs */
+
+/**
+ * Check if the specified character is in the given conversion table.
+ *
+ * @return the mapped character sequence of an ecma character, if it's in the table.
+ *         0 - otherwise.
+ */
+static ecma_length_t
+search_in_conversion_table (ecma_char_t character,        /**< code unit */
+                            ecma_char_t *output_buffer_p, /**< [out] buffer for the result characters */
+                            const ecma_char_t *array,     /**< array */
+                            const uint8_t *counters)      /**< case_values counter */
+{
+  int end_point = 0;
+  for (int i = 0; i < 3; i++)
+  {
+    int start_point = end_point;
+    int size_of_case_value = i + 1;
+    end_point += (counters[i] * (size_of_case_value + 1));
+
+    int bottom = start_point;
+    int top = end_point - size_of_case_value;
+
+    while (bottom <= top)
+    {
+      int middle = (bottom + top) / 2;
+
+      middle -= ((middle - bottom) % (size_of_case_value + 1));
+
+      ecma_char_t current = array[middle];
+
+      if (current == character)
+      {
+        if (size_of_case_value == 1)
+        {
+          output_buffer_p[0] = array[middle + 1];
+          return 1;
+        }
+        else if (size_of_case_value == 2)
+        {
+          output_buffer_p[0] = array[middle + 1];
+          output_buffer_p[1] = array[middle + 2];
+          return 2;
+        }
+        else
+        {
+          JERRY_ASSERT (size_of_case_value == 3);
+          output_buffer_p[0] = array[middle + 1];
+          output_buffer_p[1] = array[middle + 2];
+          output_buffer_p[2] = array[middle + 3];
+
+          return 3;
+        }
```

Comment:
You could use a switch here with "fallthrough":

    switch(size_of_case_value)
    {
      case 3:
        ...
        /* FALLTHRU */
      case 2:
        ...
        /* FALLTHRU */
      default:
        ...
    }